lib/widgets/message_list.dart

```diff
@@ -250,6 +247,30 @@ class _MessageListState extends State<MessageList> with PerAccountStoreAwareStat
       // TODO: Offer `ScrollViewKeyboardDismissBehavior.interactive` (or
       //   similar) if that is ever offered:
       //     https://github.com/flutter/flutter/issues/57609#issuecomment-1355340849
+      findChildIndexCallback: (Key key) {
+        // To preserve state across rebuilds for individual [MessageItem]
+        // widgets as the size of [MessageListView.items] changes we need
+        // to match old widgets by their key to their new position in
+        // the list.
+        //
+        // The keys are of type [ValueKey] with a value of [Message.id]
+        // and here we use a O(log n) binary search method. This could
+        // be improved but for now it only triggers for materialized
+        // widgets. As a simple test, scrolling through All Messages in
+        // CZO on a Pixel 5, this only runs about 10 times per rebuild
+        // and the timing for each call is measured in microseconds.
```

Comment:
> microseconds

Great!

Well, hm — does that mean "a handful of microseconds" or does it mean "less than a millisecond"?

The latter is something people might often mean by "measured in microseconds", and in this context it wouldn't actually be very encouraging — 10 times nearly-a-ms would be nearly 10ms, which would be a real problem. In general "measured in $unit" is not a very clear way of saying how big something is. For example, physicists routinely measure both subatomic and intergalactic distances in meters.

Better to be more concrete: like "<100 microseconds" or "<10 microseconds", or whatever you can most easily measure that's still specific enough to be reassuring.

Reply:
Specified that it was "<100 microseconds". The total impact would be around 1ms so it's quite reassuring the current behavior will be okay. Also changed "scrolling" to "flinging".